### PR TITLE
Issue 43

### PR DESCRIPTION
**Pull request by ronnno**
Fire DelegatedStakeChanged and Delegated events as per #43 

See last question to @OdedWx  still open

### CODE REVIEW

**Comment by ronnno** [comment]
currently still failing ` should distribute fees to validators in general and compliance committees`

File: test/delegations.spec.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        expect(r).to.have.a.delegatedStakeChangedEvent({
+            addr: p1.address,
+            selfStake: bn(100),
+            delegatedStake: bn(100),
+            delegators: [p1.address],
+            delegatorTotalStakes: [bn(100)]
+        });
+
+        const p2 = d.newParticipant();
+        r = await p2.delegate(p1);
```
Comment (vistra) [inline test/delegations.spec.ts:67]: This should also emit `delegatedStakeChangedEvent` right? `p2` has zero stake, so maybe it doesn't have to emit.

**Comment by ronnno** [comment]
@OdedWx  two questions came up:
1. There is not indication for self delegation in `DelegatedStakeChanged`, we do not count a non self delegating delegate as having a 0 amount delegated so for clarity suggesting to add a bool "selfDelegating" to this event
1. as per last comment in #43 : do we include the delegate in the delegators array (if they happen to be there)

File: contracts/Delegations.sol [hunk 2]
```diff
@@ -31,41 +31,101 @@ contract Delegations is IDelegations, IStakeChangeNotifier, ContractRegistryAcce
...
-		require(_stakeOwners.length == _updatedStakes.length, "_stakeOwners, _updatedStakes - array length mismatch");
+		uint batchLength = _stakeOwners.length;
+		require(batchLength == _amounts.length, "_stakeOwners, _amounts - array length mismatch");
+		require(batchLength == _signs.length, "_stakeOwners, _signs - array length mismatch");
+		require(batchLength == _updatedStakes.length, "_stakeOwners, _updatedStakes - array length mismatch");
 
 		_stakeChangeBatch(_stakeOwners, _amounts, _signs);
-		//TODO? emit DelegatedStakeChanged(address addr, uint256 selfSstake, uint256 delegatedStake);
+
+		emitDelegatedStakeChangedBatch(_stakeOwners, _updatedStakes);
```
[the remark this answers is not in the document]
Comment (ronnno) [reply contracts/Delegations.sol]: addressed

File: test/delegations.spec.ts (hunk 1 above)
Comment (ronnno) [reply test/delegations.spec.ts:67]: right - it will not fire the event. added to test.

**Pull request merged**